src/interface.jl

```diff
@@ -423,8 +423,10 @@ When `fallback` is true, and the obj does not have an extent,
 an extent is calculated from the coordinates of all geometries in `obj`.
 """
 function extent(obj; fallback=true)
-    ex = extent(trait(obj), obj)
-    isnothing(ex) && fallback && return calc_extent(trait(obj), obj)
+    t = trait(obj)
+    isnothing(t) && return Extents.extent(obj)
```

Comment:
Can we document the possibilities here? Before we probably errored, when the trait was `nothing`, but now the object is passed to Extents, which has its own fallback(?) to nothing?
Does it make sense to just return nothing instead ourselves, as we're asking `GeoInterface.extent(obj)` for an extent, not `Extents.extent(obj)`?

Reply:
They both return and `Extent` or `nothing`, and packages can depend on Extents.jl and not GeoInterface.jl

I can't see why we wouldn't fall back to the Extents.jl method

Reply:
I have never implemented Extents.Extent directly, only via GeoInterface.extent, so this behavior was surprising to me.

I think this is mostly a question of design and clarity, if you use GeoInterface, you either get something if a package implements GeoInterface, or nothing. In select cases, we can provide a fallback by calculating something. Here you propose to also fallback to other packages (which can return even other things, or error?).

Essentially, how to combine the GeoInterface world with the non-GeoInterface world? As you know, I have a slight preference for the world where more packages implement GeoInterface, and we extend from having just `isgeometry` to GeoInterface.issomethingthatsupportspartsofGeoInterface(Type{x}) == true. Like `israster,` or even just `is`/`hasspatial`, so we can nicely dispatch on extent, crs and the like.

Reply:
Currently YAXArrays and Rasters get `Extents.extent` for free via DimensionalData.jl dependency. YAXArrays does not have a GeoInterface dependency.

`Tyler.Map` also defines `Extents.extent` directly:
https://github.com/MakieOrg/Tyler.jl/blob/9d32abe9f25a77d774f6b5d49f2f24667ef0e496/src/Tyler.jl#L179

My design goal is to make it easy to pass as wide as possible variety of objects to any GeoInterace related function that needs an extent/bounding box, facilitated by `GeoInterface.extent` either retrieving or calculating extents for geometries and other GeoInterface impelemting objects, or falling back to `Extents.extent` for other packages that do not implement GeoInterface.

This is easier than saying "Just implement GeoInterface". Because Extents.jl is not just some random "other package", its the package that defines the `Extent` object that we already depend on, and the change is tiny to make here as is clear in this PR.

Saying that packages either implement `GeoInterface.extent` or you get `nothing` rather than the known `Extent` they return from `Extents.extent` seems a little pedantic to me. It also means packages have to implement both `Extents.extent` and `GeoInterface.extent`. 

For some code, my specific need for this and the other extents PR is to reduce this extent handling mess:
https://github.com/rafaqz/Rasters.jl/blob/main/src/utils.jl#L252-L318

Where you can see both `Extents.extent` and `GeoInterface.extent` used on different objects.